Source/monster.cpp

```diff
@@ -4799,24 +4791,24 @@ bool LineClear(bool (*Clear)(int, Point), int entity, int x1, int y1, int x2, in
 			dincH = 2 * (dx + dy);
 			yincD = -1;
 		}
-		while (!done && (x1 != x2 || y1 != y2)) {
+		while (!done && (startPoint.x != endPoint.x || startPoint.y != endPoint.y)) {
 			if ((d <= 0) ^ (yincD < 0)) {
 				d += dincD;
 			} else {
 				d += dincH;
-				y1 += yincD;
+				startPoint.y += yincD;
 			}
-			x1++;
-			done = ((x1 != xorg || y1 != yorg) && !Clear(entity, { x1, y1 }));
+			startPoint.x++;
+			done = ((startPoint.x != xorg || startPoint.y != yorg) && !Clear(entity, startPoint));

```

Comment:
Would look nicer as:
```suggestion
			done = startPoint != origin && !Clear(entity, startPoint);
```